dbms/src/Encryption/AESCTRCipherStream.cpp

```diff
@@ -29,253 +30,50 @@ namespace ErrorCodes
-                "Unexpected cipher output size for first block, expected {} actual {}",
-                block_size,
-                output_size);
-#if USE_GM_SSL
-        }
-#endif
-        memcpy(data, partial_block + block_offset, partial_block_size);
-        data_offset += partial_block_size;
-        remaining_data_size -= partial_block_size;
-    }
-
-    // Handle full blocks in the middle.
-    if (remaining_data_size >= block_size)
-    {
-        size_t actual_data_size = remaining_data_size - remaining_data_size % block_size;
-        unsigned char * full_blocks = reinterpret_cast<unsigned char *>(data) + data_offset;
-#if USE_GM_SSL
-        if (cipher_ == nullptr)
-        {
-            if (is_encrypt)
-            {
-                sm4_ctr_encrypt(&sm4_key_, iv, full_blocks, actual_data_size, full_blocks);
-            }
-            else
-            {
-                sm4_ctr_decrypt(&sm4_key_, iv, full_blocks, actual_data_size, full_blocks);
-            }
-        }
-        else
-        {
-#endif
-            ret = EVP_CipherUpdate(ctx, full_blocks, &output_size, full_blocks, static_cast<int>(actual_data_size));
-            RUNTIME_CHECK_MSG(ret == 1, "Cipher failed for offset {}.", file_offset + data_offset);
-            RUNTIME_CHECK_MSG(
-                output_size == static_cast<int>(actual_data_size),
-                "Unexpected cipher output size for block, expected {} actual {}",
-                actual_data_size,
-                output_size);
-#if USE_GM_SSL
-        }
-#endif
-        data_offset += actual_data_size;
-        remaining_data_size -= actual_data_size;
-    }
-
-    // Handle partial block at the end. The partial block is copied to buffer to
-    // fake a full block.
-    if (remaining_data_size > 0)
-    {
-        assert(remaining_data_size < AES_BLOCK_SIZE);
-        memcpy(partial_block, data + data_offset, remaining_data_size);
-#if USE_GM_SSL
-        if (cipher_ == nullptr)
-        {
-            if (is_encrypt)
-            {
-                sm4_ctr_encrypt(&sm4_key_, iv, partial_block, block_size, partial_block);
-            }
-            else
-            {
-                sm4_ctr_decrypt(&sm4_key_, iv, partial_block, block_size, partial_block);
-            }
-        }
-        else
-        {
-#endif
-            ret = EVP_CipherUpdate(ctx, partial_block, &output_size, partial_block, block_size);
-            RUNTIME_CHECK_MSG(ret == 1, "Cipher failed for last block, offset {}.", file_offset + data_offset);
-            RUNTIME_CHECK_MSG(
-                output_size == static_cast<int>(block_size),
-                "Unexpected cipher output size for last block, expected {} actual {}",
-                block_size,
-                output_size);
-#if USE_GM_SSL
-        }
-#endif
-        memcpy(data + data_offset, partial_block, remaining_data_size);
-    }
-#endif
+    DB::Encryption::Cipher(file_offset, data, data_size, key, method, iv, /*is_encrypt=*/true);
```

Comment:
In the master implementation, `AESCTRCipherStream` use the same `cipher` instance for all `encryption` and `decryption` calls. But now it create a new `cipher` instance for each `encryption` and `decryption` calls.
Will it bring different encryption result? Will it bring performance issue?

Reply:
It just return a pointer which point to a static instance, no new instance is created.